### PR TITLE
Enhanced RDoc for Array

### PR DESCRIPTION
**Pull request by BurdetteLamar**
Methods:
- keep_if
- delete
- delete_at
- slice!

### CODE REVIEW

File: array.c [hunk 1]
```diff
@@ -4292,22 +4302,42 @@ ary_resize_smaller(VALUE ary, long len)
...
- *     ary.delete(obj) {block}    -> item or result of block
+ *    array.delete(obj) -> deleted_object
+ *    array.delete {|obj| ... } -> deleted_object or block_return
+ *
+ *  Removes zero or more elements from +self+; returns +self+.
+ *
+ *  ---
+ *
+ *  When no block is given,
+ *  removes from +self+ each element +ele+ such <tt>that ele == obj</tt>;
```
Comment (drbrain) [inline array.c:4313]: I think `that` jumped inside the `<tt>` here

Comment (BurdetteLamar) [reply array.c:4313]: _That_s are like that!

File: array.c [hunk 2]
```diff
@@ -4292,22 +4302,42 @@ ary_resize_smaller(VALUE ary, long len)
 
 /*
  *  call-seq:
- *     ary.delete(obj)            -> item or nil
- *     ary.delete(obj) {block}    -> item or result of block
+ *    array.delete(obj) -> deleted_object
+ *    array.delete {|obj| ... } -> deleted_object or block_return
```
Comment (drbrain) [inline array.c:4306]: This should be `array.delete(obj) {|nosuch| … }` (or replace `nosuch` with another name as Array#delete requires an argument

File: array.c [hunk 3]
```diff
@@ -4292,22 +4302,42 @@ ary_resize_smaller(VALUE ary, long len)
...
+ *  Returns +nil+ if no elements removed:
+ *    a = [:foo, 'bar', 2]
+ *    a.delete(:nosuch) # => nil
  *
- *  Deletes all items from +self+ that are equal to +obj+.
+ *  ---
  *
- *  Returns the last deleted item, or +nil+ if no matching item is found.
+ *  When a block is given,
+ *  removes from +self+ each element +ele+ such <tt>that ele == obj</tt>.
```
Comment (drbrain) [inline array.c:4328]: Same here with `that`

File: array.c [hunk 4]
```diff
@@ -4387,18 +4417,43 @@ rb_ary_delete_at(VALUE ary, long pos)
 }
 
 /*
- *  call-seq:
- *     ary.delete_at(index)  -> obj or nil
+ *  call-seq
+ :
```
Comment (drbrain) [inline array.c:4421]: This `:` jumped to a new line

File: array.c (hunk 1 above)
Comment (BurdetteLamar) [reply array.c:4313]: Fixed in next PR.

File: array.c (hunk 3 above)
Comment (BurdetteLamar) [reply array.c:4328]: Fixed in next PR.

File: array.c (hunk 4 above)
Comment (BurdetteLamar) [reply array.c:4421]: Fixed in next PR.

File: array.c (hunk 2 above)
Comment (BurdetteLamar) [reply array.c:4306]: Fixed in next PR.

File: array.c [hunk 5]
```diff
@@ -4303,14 +4303,14 @@ ary_resize_smaller(VALUE ary, long len)
 /*
  *  call-seq:
  *    array.delete(obj) -> deleted_object
- *    array.delete {|obj| ... } -> deleted_object or block_return
+ *    array.delete {|nosuch| ... } -> deleted_object or block_return
```
Comment (drbrain) [inline array.c:4306]: This line still needs `(obj)` after `delete`:

```suggestion
 *    array.delete(obj) {|nosuch| ... } -> deleted_object or block_return
```

Comment (drbrain) [reply array.c:4306]: After ↑ it looks good to merge

**Pull request merged**